examples/src/main/scala/zio/config/examples/ReadWriteReport.scala

```diff
@@ -1,63 +1,57 @@
 package zio.config.examples
 
 import zio.DefaultRuntime
-import zio.config._
+import zio.config._, Config._
+import zio.config.actions.UserManual
+import zio.config.actions.UserManual.KeyDescription
 
-/**
- * An example that shows the usage of pretty much all existing combinators.
- * <+> , |@|, | etc*
- *
- * It also shows:
- *  1) How documentation is done.
- *  2) How config is Read
- *  3) How config is Written back
- *  4) How to only Report config
- */
 object ReadWriteReport extends App {
 
   case class Password(value: String)
-  case class UserPwd(name: String, pwd: Option[Password], anonymouse: Option[String])
+  case class UserPwd(name: String, pwd: Option[Password], abc: Option[String], value: Option[XYZ])
   case class Token(value: String, clientid: String)
+  case class XYZ(xyz: String, someInteger: Int)
 
   type ProdConfig = Either[UserPwd, Token]
 
-  import Config._
-
   // An example where user provides a description once and for all, and use it for read, write, report!
-  val config: Config[ProdConfig] =
-    (string("user") <*> string("pwd").xmap(Password)(_.value).optional <*> string("anonymous").optional)(
+  val config: ConfigDescriptor[ProdConfig] =
+    ((string("usr") ~ "Example: some-user" <*>
+      string("pwd").xmap(Password)(_.value).optional ~ "We don't care" ~ "yea !" <*>
+      string("jhi").optional ~ "Example: ghi" <*>
+      (string("xyz") <*> int("abc"))(XYZ.apply, XYZ.unapply).optional ~ "Example: xyz")(
       UserPwd.apply,
       UserPwd.unapply
     ) or
-      (string("auth_token") <*> string("clientid"))(Token.apply, Token.unapply)
+      (string("auth_token") <*> string("clientid"))(Token.apply, Token.unapply)) ~ "Prod Config"
 
```

Comment:
The initial description and comparability was tested in example. We will have a task to set up test cases for man-page/usermanual.

Reply:
We can see we were able to tag /document the entire config as `Prod Config`, and each individual ones as well, stacked together when creating man page. :)

Reply:
Also optional values are automatically documented as optional. Please see below for the results